python/pyspark/ml/functions.py

```diff
@@ -14,9 +14,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from __future__ import annotations
 
+import inspect
+import numpy as np
+import pandas as pd
+import uuid
 from pyspark import SparkContext
+from pyspark.sql.functions import pandas_udf
 from pyspark.sql.column import Column, _to_java_column
+from pyspark.sql.types import (
+    ArrayType,
+    ByteType,
+    DataType,
+    DoubleType,
+    FloatType,
+    IntegerType,
+    LongType,
+    ShortType,
+    StringType,
+    StructType,
+)
+from typing import Any, Callable, Iterator, List, Mapping, Protocol, TYPE_CHECKING, Tuple, Union
+
+if TYPE_CHECKING:
+    from pyspark.sql._typing import UserDefinedFunctionLike
+
+supported_scalar_types = (
+    ByteType,
+    ShortType,
+    IntegerType,
+    LongType,
+    FloatType,
+    DoubleType,
+    StringType,
+)
+
+
+class PredictFunction(Protocol):
```

Comment:
Since this is a public interface now. Shall we call it `PredictBatchFunction` to make it clear it is paired with `predict_batch_udf`?